src/main/java/net/minecraftforge/server/permission/PermissionAPI.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.server.permission;
+
+import com.google.common.base.Preconditions;
+import com.mojang.authlib.GameProfile;
+import net.minecraftforge.fml.common.FMLLog;
+import net.minecraftforge.server.permission.context.IContext;
+import org.apache.logging.log4j.Level;
+
+import javax.annotation.Nonnull;
+
+public class PermissionAPI
+{
+    private static IPermissionHandler permissionHandler = DefaultPermissionHandler.INSTANCE;
+
+    public static void setPermissionHandler(@Nonnull IPermissionHandler handler)
```

Comment:
The idea being that you can just plop a perms handling mod into your mods folder and you're golden, which makes it far more accessible for the user. Sponge already does this anyway https://docs.spongepowered.org/master/en/plugin/services.html

I understand where you're coming from, because this has a potential for abuse, but I can't agree with this.

What I can suggest for the ability for the user to force an override in the config, working similarly to what you said. If the user has multiple perm providers installed, it will either take the last registered one, or if an override is specified, it will use the selected provider.